optimum/onnxruntime/modeling_seq2seq.py

```diff
@@ -611,10 +717,37 @@ class ORTModelForSeq2SeqLM(ORTModelForConditionalGeneration, GenerationMixin):
     Sequence-to-sequence model with a language modeling head for ONNX Runtime inference.
     """
 
+    export_feature = "seq2seq-lm"
+    auto_model_class = AutoModelForSeq2SeqLM
+
     def __init__(self, *args, **kwargs):
-        super().__init__(*args, **kwargs)
         self.main_input_name = "input_ids"
 
+        super().__init__(*args, **kwargs)
+
+    def _initialize_encoder(self, encoder_session: onnxruntime.InferenceSession, device: str) -> ORTEncoder:
+        return ORTEncoder(session=encoder_session, device=device, main_input_name=self.main_input_name)
+
+    def get_encoder_onnx_config(
+        encoder_model: "PreTrainedModel", onnx_config: OnnxSeq2SeqConfigWithPast = None
```

Comment:
The functions were added to prevent any hardcoding based on model type in the `ORTModelForConditionalGeneration._from_transformers()` function for retrieving the onnx configs for encoder and decoder.

For the seq2seq models such as T5 the onnx_configs are generated using local Encoder and Decoder configs, hence this variable becomes unused.

Reply:
Not relevant